### ISSUE TITLE
[HOLD for payment 2024-06-20][Unvalidated Sign-Up] [$500] Add validation flow to bank account set up 

### ISSUE THREAD

**Issue by NikkiWines** [Expensify/App#33035]
## Action Performed:
1. Create an unvalidated account on New Expensify by logging into expensify.com with an account on a public domain
2. Create a workspace for that account 
3. Navigate to your Workspace > 
4. Select either `Connect Online With Plaid` or `Connect Manually` 

## Expected Result:
5. User should see a page with a view to validate their account and receive a notification to validate their bank account. 
     - We can use the existing `RequestContactMethodValidateCode` command for this 
7. Pressing the `<` should take them back to the `Connect Bank Account` with the two options mentioned in step 4. 
8. Once the user verifies their account they should be able to continue with the standard bank account setup flow 
     - Verifying the account can be done via the existing `ValidateLogin` command, though the navigation will need to be updated so that the user is not routed away from the bank account flow. 

![Untitled](https://github.com/Expensify/App/assets/16747822/96a5c604-b07d-481b-aebd-70c73edcb115)

## Actual Result:
User is blocked from adding a bank account by an error message. 

<img width="300" alt="" src="https://github.com/Expensify/App/assets/16747822/4fb5d800-9cbe-4dea-afa6-68031e96ada5">

## Workaround:
N/A 

## Platforms:
Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0160a09bc15c94c626</li>
        <li>Upwork Job ID: 1735348005410426880</li>
        <li>Last Price Increase: 2023-12-14</li>
<li>Automatic offers: </li>
<ul>
<li>dukenv0307 | Contributor | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @lschurr</details>

**Comment by shubham1206agra** [comment]
@mountiny Please also mark this as hold on VBBA refactor

**Comment by mountiny** [comment]
Adding this on hold for the bank account flow refactor it will be a major change https://github.com/Expensify/App/issues/30442

**Comment by lschurr** [comment]
On hold.

**Comment by lschurr** [comment]
On hold.

**Comment by lschurr** [comment]
On hold.

**Comment by lschurr** [comment]
On hold.

**Comment by lschurr** [comment]
On hold.

**Comment by lschurr** [comment]
On hold.

**Comment by lschurr** [comment]
Hold.

**Comment by lschurr** [comment]
On hold.

**Comment by lschurr** [comment]
Removing hold. What's next @shubham1206agra @allroundexperts?

**Comment by allroundexperts** [comment]
Hi @dukenv0307!
Can you please update your proposal based on the VBBA refactor?

**Comment by allroundexperts** [comment]
Right, I think we need some backend help here.

**Comment by allroundexperts** [comment]
@dukenv0307's [proposal](https://github.com/Expensify/App/issues/33035#issuecomment-1855394934) looks good to me. However, as stated [here](https://github.com/Expensify/App/issues/33035#issuecomment-1985836551), we'll need some backend help to implement this.

🎀 👀 🎀 C+ reviewed

**Comment by lschurr** [comment]
We'll need to attach this to a wave/vip project I think and get an internal engineer to volunteer. I'll work on that later today.

**Comment by lschurr** [comment]
Asking here - https://expensify.slack.com/archives/C01SKUP7QR0/p1710782144083309

**Comment by NikkiWines** [comment]
Hmm, yeah. Looks like we should probably make a new command for this, as ultimately we kind of want a hybrid of `User.validateSecondaryLogin()` and `Session.signIn()`, and neither quite fit as they're written right now. 

Maybe something called `Session.validateLoginAndSignIn()`? We should probably be using that for the anonymous account 
claiming logic as well now that I think about it 🤔  (cc: @marcochavezf) 

I'll work on adding that to the backend and update here when it's deployed

**Comment by lschurr** [comment]
Any update on this one @NikkiWines @dukenv0307?

**Comment by NikkiWines** [comment]
No update, sorry, this is waiting on me. I'm currently focusing on Release 1 projects at the moment and this is slated for Release 3. I'll try to work on the new command if I have some spare time

**Comment by lschurr** [comment]
Believe this is same as above.

**Comment by NikkiWines** [comment]
No update here

**Comment by lschurr** [comment]
No update yet on this one!

**Comment by trjExpensify** [comment]
Moving this under the umbrella of [Unvalidated Sign-Up] for release 2.

**Comment by NikkiWines** [comment]
@dukenv0307 it looks like we now set the session authToken to an updated authToken when calling `validateSecondaryLogin`, so you shouldn't need any additional backend changes for this to work 🍀

**Comment by lschurr** [comment]
Not overdue. This is still 2 days from payment.

**Comment by lschurr** [comment]
Payment summary:

- Contributor: @dukenv0307 $500 (to be [paid in Upwork](https://www.upwork.com/nx/wm/offer/101439478))
- Contributor+: @alitoshmatov $500 (please request in NewDot)

**Comment by lschurr** [comment]
All set!

**Comment by allroundexperts** [comment]
Hi @lschurr!

I actually reviewed this PR. Can you please update the payment summary? Thanks!

**Comment by lschurr** [comment]
Ooops sorry about that @allroundexperts - I think that was GH auto-populating. I updated the summary!

**Comment by JmillsExpensify** [comment]
$500 approved for @allroundexperts
